src/text-fragment-utils.js

```diff
@@ -150,6 +153,8 @@ export const processFragmentDirectives =
  * @param {TextFragment} textFragment - Text Fragment to highlight.
  * @param {Document} documentToProcess - document where to extract and mark
  *     fragments in.
+ * @param {Element} root - the root element where to extract and mark
```

Comment:
```suggestion
 * @param {Element=} root - the root element where to extract and mark
```

To mark the param as optional.